android-components/automation/application-services-nightly-bump.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/python3
+
+import json
+import gzip
+import os
+import re
+import sys
+from urllib.request import urlopen
+
+# Latest nightly maven.json file from the taskcluster index
+MAVEN_JSON_URL = "https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/project.application-services.v2.nightly.latest/artifacts/public%2Fbuild%2Fnightly%2Fmaven.json"
+
+VERSION_RE = re.compile(r'val VERSION = "[\d\.]+"')
+CHANNEL_RE = re.compile(r'val CHANNEL = ApplicationServicesChannel.[A-Z_]+')
+
+def main():
+    with open(kt_module_path(), "r+t") as f:
+        f.seek(0)
+        current_source = f.read()
+        new_source = make_replacements(current_source)
+        f.seek(0)
+        f.write(new_source)
+
+def kt_module_path():
+    root_dir = os.path.dirname(os.path.dirname(__file__))
+    return os.path.join(root_dir, "plugins/dependencies/src/main/java/ApplicationServices.kt")
+
+def make_replacements(source):
+    if VERSION_RE.search(source) is None:
+        print("Couldn't find VERSION initialization line")
+        sys.exit(1)
+    if CHANNEL_RE.search(source) is None:
+        print("Couldn't find CHANNEL initialization line")
+        sys.exit(1)
+    maven_config = fetch_maven_config()
+    version = maven_config['version']
+    if maven_config['channel'] == 'maven-release':
+        channel = "RELEASE"
+    elif maven_config['channel'] == 'maven-staging':
+        channel = "STAGING"
+    elif maven_config['channel'] == 'maven-nightly':
+        channel = "NIGHTLY"
+    elif maven_config['channel'] == 'maven-nightly-staging':
+        channel = "NIGHTLY_STAGING"
+    else:
+        print("Unknown `channel` value: {}".format(maven_config['channel']))
+        sys.exit(1)
+
+    source = VERSION_RE.sub(f'val VERSION = "{version}"', source)
+    source = CHANNEL_RE.sub(f'val CHANNEL = ApplicationServicesChannel.{channel}', source)
+    return source
+
+def fetch_maven_config():
+    response = urlopen(MAVEN_JSON_URL)
+    return json.loads(gzip.decompress(response.read()))
+
+if __name__ == '__main__':
+    main()
```

Comment:
As mentioned in the commit message, once we've agreed on everything and the app-services nightlies are running, we should:
- Run this script manually
- Once we figure out any snags to running manually, setup a cron job to run this script, create a commit and PR for it.
- Once that cron job is running smoothly, setup automerge on the PRs.

How does that sound to others?

Reply:
That sounds great to me.  So would this mean updating the `android_components._update_application_services()` function to pull in the latest nightly using the same basic process as the `application-services-nightly-bump.py` script?

Reply:
In a perfect world, it'll also be able to handle the Nightly major version change too once that bump hits Maven.

Reply:
Relbot definitely seems like the right place for this, I made this PR in that repo: https://github.com/mozilla-mobile/relbot/pull/105